modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -124,10 +127,20 @@ public void requestReceived(NHttpServerConnection conn) {
             if (isMessageSizeValidationEnabled) {
                 httpContext.setAttribute(PassThroughConstants.MESSAGE_SIZE_VALIDATION_SUM, 0);
             }
-
             SourceRequest request = getSourceRequest(conn);
             if (request == null) {
                 return;
+
+            }
+            //check correlationEnabling parameter
+            String correlationStatus = sourceConfiguration.getCorrelationStatus();
+            boolean correlationLoggingEnabled = correlationStatus.equals(PassThroughConstants.CORRELATION_ENABLE_STATE);
```

Comment:
do we need correlationLoggingEnabled variable ? can we do the check inside the next if condition ? 
Better to check as PassThroughConstants.CORRELATION_ENABLE_STATE.equals(sourceConfiguration.getCorrelationStatus()) . (constant first to avoid any NPEs)